src/Compilers/VisualBasic/Test/Emit/CodeGen/CodeGenTuples.vb

```diff
@@ -7459,6 +7459,210 @@ BC30512: Option Strict On disallows implicit conversions from 'Double' to 'Strin
+Class C(Of T, U)
+End Class
+]]></file>
+</compilation>, additionalRefs:=s_valueTupleRefs)
+
+            comp.AssertTheseDiagnostics(<errors>
+BC30311: Value of type '(Integer, Object)' cannot be converted to 'C(Of Integer, T)'.
+        Dim x = TryCast((0, Nothing), C(Of Integer, T))
+                        ~~~~~~~~~~~~
+                                        </errors>)
+
+            Dim tree = comp.SyntaxTrees.Single()
+            Dim model = comp.GetSemanticModel(tree)
+            Dim node = tree.GetRoot().DescendantNodes().OfType(Of TupleExpressionSyntax)().Single()
+            Assert.Equal("(0, Nothing)", node.ToString())
+            Assert.Null(model.GetTypeInfo(node).Type)
+            Assert.Equal("(System.Int32, System.Object)", model.GetTypeInfo(node).ConvertedType.ToTestDisplayString())
+            Assert.Equal(ConversionKind.WideningTuple, model.GetConversion(node).Kind)
+
+            Assert.Equal("C(Of System.Int32, T)", model.GetTypeInfo(node.Parent).Type.ToTestDisplayString())
+            Assert.Equal("C(Of System.Int32, T)", model.GetTypeInfo(node.Parent).ConvertedType.ToTestDisplayString())
+
+        End Sub
+
+        <Fact>
+        Public Sub TupleImplicitNullableConversionWithTypelessTuple()
+            Dim comp = CreateCompilationWithMscorlibAndVBRuntime(
+<compilation>
+    <file name="a.vb"><![CDATA[
+Option Strict On
+Class C
+    Shared Sub Main()
+        Dim x As (Integer, String)? = (1, Nothing)
+        System.Console.Write(x)
+    End Sub
+End Class
+]]></file>
+</compilation>, additionalRefs:=s_valueTupleRefs, options:=TestOptions.DebugExe)
+
+            comp.AssertNoDiagnostics()
+            CompileAndVerify(comp, expectedOutput:="(1, )")
+
+            Dim tree = comp.SyntaxTrees.Single()
+            Dim model = comp.GetSemanticModel(tree)
+            Dim node = tree.GetRoot().DescendantNodes().OfType(Of TupleExpressionSyntax)().Single()
+            Assert.Equal("(1, Nothing)", node.ToString())
+            Assert.Null(model.GetTypeInfo(node).Type)
+            Assert.Equal("System.Nullable(Of (System.Int32, System.String))", model.GetTypeInfo(node).ConvertedType.ToTestDisplayString())
+            Assert.Equal(ConversionKind.WideningNullableTuple, model.GetConversion(node).Kind)
+
+        End Sub
+
+        <Fact>
+        Public Sub TupleImplicitNullableAndCustomConversionsWithTypelessTuple()
+            Dim comp = CreateCompilationWithMscorlibAndVBRuntime(
+<compilation>
+    <file name="a.vb"><![CDATA[
+Option Strict Off
+Structure C
+    Shared Sub Main()
+        Dim x As C = (1, Nothing)
+        Dim y As C? = (2, Nothing)
+    End Sub
+    Public Shared Widening Operator CType(ByVal d As (Integer, String)) As C
+          Return New C()
+    End Operator
+End Structure
+]]></file>
+</compilation>, additionalRefs:=s_valueTupleRefs, options:=TestOptions.DebugExe)
+
+            comp.AssertTheseDiagnostics(<errors>
+BC30311: Value of type '(Integer, Object)' cannot be converted to 'C?'.
```

Comment:
Investigated this. The tuple has type `(Integer, Object)` and the conversion method has input type `(Integer, String)`. So the input would have to go through a narrowing tuple conversion, which gets rejected by `DetermineMostSpecificWideningConversion` because it is not widening.

On the other side, `DetermineMostSpecificNarrowingConversion` rejects this method because the out conversion would be widening (widening nullable). See comment below.

From the comment, I think the logic needs to account for tuples in some new way. I'm not sure how. I'll stop by tomorrow to discuss.

```VB
' Note that {Narrowing in, Widening (non-identity) out} operator is not considered as an applicable candidate.
                            ' In fact, an operator like this cannot exist unless nullable in/out conversions are involved.
                            ' Basically we would be dealing with an operator that converts from type derived from source to type derived from destination,
                            ' it would have to be defined in one of those types. When we collect operators we only visit source, destination and their
                            ' bases. So, in order for such an operator to be found, there must be an inheritance relationship between source and
                            ' destination and the operator must be defined in a type that is in between of them in the inheritance hierarchy. Thus,
                            ' there would be an inheritance relationship between parameter type ant return type of the operator, which makes the operator
                            ' inapplicable - there would be a predefined conversion between the types.
                            ' Ignoring an operator like this even when nullable conversions are involved, allows us to consider it as a candidate for a 
                            ' lifting, otherwise it would be treated as a not-lifted narrowing.                         
```